### PR TITLE
MAINT: Upgrade to anaconda=2024.02 and cuda=12.3.1

### PR DESCRIPTION
**Pull request by mmcky**
This PR

- [x] upgrades to `anaconda=2024.02`
- [x] upgrades to `CUDA=12.3.1`

this PR is part of https://github.com/QuantEcon/meta/issues/130

requires: #387 

- [x] this PR is waiting on `interpolations` to be updated for `numba>=0.59`

### CODE REVIEW

**Comment by mmcky** [comment]
- [ ] the broken link is to `BK19` as a reference but that doesn't seem to be in our `quant-con.bib` file 

```
You can read  about Dynamic Mode Decomposition here {cite}`DMD_book` and here [[BK19](https://python.quantecon.org/zreferences.html#id25)] (section 7.2).
```

@jstac or @thomassargent30 do you know what BK19 would refer to and I can add it as a proper reference using `bibtex`

**Comment by jstac** [comment]
@mmcky , it refers to this one: 
https://www.databookuw.com/

**Comment by mmcky** [comment]
- [ ] this pr requires #387

**Comment by mmcky** [comment]
- [ ] this cannot be merged until `interpolations` is updated or there is a substitute for `mlinterp` that is suitable

```
ImportError[0m: cannot import name 'generated_jit' from 'numba' (/opt/conda/envs/quantecon/lib/python3.11/site-packages/numba/__init__.py)
```

**Comment by kp992** [comment]
I see, from numba `0.60`, `generated_jit` will fail.

**Comment by mmcky** [comment]
> I see, from numba `0.60`, `generated_jit` will fail.

@kp992 I think from `numba>=0.59` it will fail -- so that is why we will need to fix `interpolation` as we use it for `minterp` and some splines support. From my discussion with Pablo, the `generated_jit` will need to be replaced with an alternative (`overload`?) https://github.com/EconForge/interpolation.py/issues/110

**Comment by kp992** [comment]
> From my discussion with Pablo, the `generated_jit` will need to be replaced with an alternative (`overload`?)

I have tried fixing that locally but that's throwing some error. I followed the examples of moving from `generated_jit` to `overload`, but didn't work. Let's pin `numba < 0.59` until we find and do robust a fix for this issue.

**Comment by mmcky** [comment]
- [ ] check status of https://github.com/EconForge/interpolation.py/pull/114

File: lectures/odu.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ In addition to what’s in Anaconda, this lecture deploys the libraries:
 ---
 tags: [hide-output]
 ---
-!pip install interpolation
+!pip install git+https://github.com/EconForge/interpolation.py
```
Comment (mmcky) [inline lectures/odu.md]: @jstac this installs `interpolation` package in the `odu` lecture from the `main` branch of the repo as it isn't released to pypi yet. Are you happy to run with this until `pypi` is updated?

**Comment by mmcky** [comment]
- [ ] re-enable build cache
- [ ] open an issue to update pip install from `main` branch of `interpolation` repo to pypi once released

File: lectures/odu.md (hunk 1 above)
Comment (mmcky) [inline lectures/odu.md]:
```suggestion
!pip install interpolation
```

**Comment by mmcky** [comment]
- [x] reverted to install `interpolation` from `pypi` which is currently `2.2.6`

**Pull request merged**